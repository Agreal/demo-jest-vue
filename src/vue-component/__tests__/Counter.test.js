import { mount } from '@vue/test-utils'
import Counter from '../Counter.vue'

describe('Counter', () => {

  describe('render', function () {
    const wrapper = mount(Counter);

    test('the correct markup', () => {
      expect(wrapper.html()).toContain('<span class="count">0</span>');
    });

    test('has a button', () => {
      expect(wrapper.contains('button')).toBe(true);
    });

    test('correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    })
  });

  describe('Click event', function () {
    test('should increase count', () => {
      const wrapper = mount(Counter);
      wrapper.find('button').trigger('click');
      expect(wrapper.html()).toContain('<span class="count">1</span>');
    });

    test('render correctly', () => {
      const wrapper = mount(Counter);
      wrapper.find('button').trigger('click');
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should should increase count', () => {
      const wrapper = mount(Counter);
      wrapper.find('button').trigger('click');
      expect(wrapper.vm.count).toBe(1);
    });
  });
});
