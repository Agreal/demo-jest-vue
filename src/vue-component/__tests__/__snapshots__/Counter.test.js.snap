// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Click event render correctly 1`] = `
<div>
  <span
    class="count"
  >
    1
  </span>
   
  <button>
    Increment
  </button>
</div>
`;

exports[`Counter render correctly 1`] = `
<div>
  <span
    class="count"
  >
    0
  </span>
   
  <button>
    Increment
  </button>
</div>
`;
